glyph-brush-layout/src/lines.rs

```diff
@@ -102,9 +102,8 @@ impl<'font, L: LineBreaker, F: FontMap<'font>> Iterator for Lines<'_, '_, 'font,
         let mut progressed = false;
 
         while let Some(word) = self.words.peek() {
-            let word_max_x = word.bounds.map(|b| b.max.x).unwrap_or(word.layout_width);
             // only if `progressed` means the first word is allowed to overlap the bounds
-            if progressed && (caret.x + word_max_x).ceil() > self.width_bound {
+            if progressed && caret.x + word.layout_width_no_trail > self.width_bound {
```

Comment:
This has clearer intent now too. I'm not sure what my idea was with the `ceil` before.

It's worth noting that using glyph_bounds as bounds can still fail due to floating point accuracy loss. I'm not sure if it's worth trying to deal with that here, ie allow relative equality.

Reply:
I've reproduced and fixed the fp error case in #91